### PR TITLE
feat: its example

### PR DESCRIPTION
**Pull request by Foivos**
[AXE-4712]

[AXE-4712]: https://axelarnetwork.atlassian.net/browse/AXE-4712?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: move/example/sources/its/its.move [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        arguments.push_back(arg);
+
+        // ITS object
+        arg = vector::singleton<u8>(0);
+        arg.append(object::id_address(its).to_bytes());
+        arguments.push_back(arg);
+
+        // payload
+        arg = vector[ 3 ];
+        arguments.push_back(arg);
```
Comment (npty) [inline move/example/sources/its/its.move:58]: create a helper function to construct each argument instead? 

something like this
```move
fun create_argument<T>(prefix: u8, value: T): vector<u8> {
   let mut arg = vector::singleton<u8>(prefix);
   arg.append(object::id_address(value).to_bytes());
   arg 
}
```

then, maybe reuse it like below

```
let arguments = vector[
  create_argument(0, singleton),
  create_argument(0, its),
  vector[3],
];
```

and we can reuse it in the `get_transaction` function as well.

File: move/example/sources/its/its.move [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+            id: singletonId,
+            channel,
+        });
+    }
+
+    public fun register_transaction(discovery: &mut RelayerDiscovery, singleton: &Singleton, its: &ITS) {
+        let mut arguments = vector::empty<vector<u8>>();
+
+        // Singleton object
+        let mut arg = vector::singleton<u8>(0);
```
Comment (npty) [inline move/example/sources/its/its.move:47]: perhaps we create a module somewhere to store all possible argument types?

for example,

```move
module gmp::arg_types {
public const TYPE_OBJECT u8 = 0;
/// Prefix for pure arguments, followed by the BCS encoded form of the pure
public const TYPE_PURE u8 = 1;
/// ... and so on
```

then import and use it here like below:
```move
use gmp::arg_types;

create_argument(arg_types::TYPE_OBJECT, singleton)
```

File: .github/workflows/test-js.yaml [hunk 3]
```diff
@@ -33,6 +33,6 @@ jobs:
       - name: Sleep for 30 seconds
         run: sleep 30s
         shell: bash
-
+
```
Comment (milapsheth) [inline .github/workflows/test-js.yaml:36]: extra whitespace

File: move/example/sources/its/its.move [hunk 4]
```diff
@@ -0,0 +1,211 @@
+module example::its_example {
```
Comment (milapsheth) [inline move/example/sources/its/its.move:1]: try adding the move formatter and run again (ping mysten team if there's an issue, maybe they have a newer version)

File: move/example/sources/its/its.move [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+        discovery::register_transaction(discovery, &singleton.channel, transaction);
+    }
+
+    public fun get_transaction(singleton: &Singleton, its: &ITS, payload: vector<u8>): Transaction {
+        let mut arguments = vector::empty<vector<u8>>();
+
+        // ApprovedMessage
+        let mut arg = vector::singleton<u8>(2);
+        arguments.push_back(arg);
+
```
Comment (milapsheth) [inline move/example/sources/its/its.move:115]:
```suggestion
```

File: move/example/sources/its/its.move (hunk 4 above)
Comment (milapsheth) [inline move/example/sources/its/its.move:1]: Add docstring explaining how the ITS example works

File: test/its.js [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+    let discovery;
+    let its;
+    let axelarPackageId;
+    let exampleId, singleton, singletonChannel;
+    let governance;
+    const remoteChain = 'Remote Chain';
+    const trustedAddress = 'Trusted Address';
+    const gatewayInfo = {};
+
+    function calculateNextSigners() {
```
Comment (milapsheth) [inline test/its.js:43]: avoid code duplication

File: test/its.js [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+        let result = await publishPackage(client, deployer, 'axelar_gateway');
+        axelarPackageId = result.packageId;
+        const creatorCap = result.publishTxn.objectChanges.find(
+            (change) => change.objectType === `${axelarPackageId}::gateway::CreatorCap`,
+        ).objectId;
+        discovery = result.publishTxn.objectChanges.find(
+            (change) => change.objectType === `${axelarPackageId}::discovery::RelayerDiscovery`,
+        ).objectId;
+
+        await publishPackage(client, deployer, 'abi');
```
Comment (milapsheth) [inline test/its.js:106]: create a deployITS helper?

File: test/its.js [hunk 8]
```diff
@@ -0,0 +1,260 @@
...
+    },
+} = require('../dist/bcs');
+const { bcs } = require('@mysten/sui/bcs');
+const {
+    utils: { arrayify, hexlify, keccak256, defaultAbiCoder },
+    constants: { HashZero },
+} = require('ethers');
+
+const clock = '0x6';
+const sui = '0x2';
```
Comment (milapsheth) [inline test/its.js:19]: reuse constants instead of hardcoding

File: test/its.js [hunk 9]
```diff
@@ -0,0 +1,260 @@
...
+const MESSAGE_TYPE_SET_TRUSTED_ADDRESSES = '0x2af37a0d5d48850a855b1aaaf57f726c107eb99b40eabf4cc1ba30410cfa2f68';
```
Comment (milapsheth) [inline test/its.js:20]: hash instead of hardcoding

**Review by milapsheth — APPROVED** [review]

**Pull request closed without merging**